core/src/main/scala/api/sfs.scala

```diff
@@ -25,15 +30,19 @@ trait Filesystem {
    *  things too eagerly, nor caching data for too long.
    */
   def resolve(path: Path): Key
-  def metadata(key: Key): Metadata
-  def lookup(key: Key): Data
+  def metadata(key: Key): M[Metadata]
+  def lookup(key: Key): M[Data]
```

Comment:
I have been thinking about that as well and I like that idea.

What I'm troubled with, is how to state metadata requirements. It will be a constant source / documentation lookup if we can not make clear what is supposed / expected to be present in the metadata for something to work.

I'm thinking something along the lines of:

```
val underlying: FileSystem {
  type RecognisedMetadata = Size :+: Atime :+: ...
}
```